languages/csharp/exercises/concept/parameters/.docs/instructions.md

```diff
@@ -0,0 +1,80 @@
+The remote control car project you kicked off in the `classes` exercise has gone well (congratulations!) and due to a number of recent sponsorship deals there is money in the budget for enhancements.
+
+Part of the budget is being used to provide some telemetry.
+
+To keep the sponsors sweet a panel has been installed on the car to display the sponsors names as it goes round the track.
+
+You will note that the introduction of these fancy new technologies has dramatically reduced the car's battery life.
+
+## 1. Enable the remote control car to display sponsor names
+
+Implement `AddSponsors` to take one or more sponsor names and store them on the car.
```

Comment:
I had some trouble figuring out why the `AddSponsors` method bothered me a bit, but I think I know now. It being a `params` parameter also means you could do `AddSponsors()` (without any values). I came up with two alternatives:

1. Have the `Buy` method take a `params string[] sponsors`  parameter and then pass that on to the constructor. This means having to modify the constructor to to the sponsors as a parameter.
1. Rename the method to `SetSponsors` to indicate that it will replace all the existing sponsors. Then you also don't need the `is guaranteed to be called at most once` constraint.

Personally, I prefer the second option. Let me know what you think.

Reply:
Agreed